### PR TITLE
Always include native inet_aton

### PR DESCRIPTION
**Pull request by sbc100**
This file was only being included under some circumstances
but its trivial wrapper around inet_pton.c which was always
being included in libc anyway.

This change helps with the musl upgrade as it means we can
build libsockets_proxy without the musl internal include
paths enabled.

### CODE REVIEW

**Comment by sbc100** [comment]
It looks like this was probably a codesize thing.

The native version pulls in strtol which adds about 2k to the overall code size of a trivial program that simple references this symbol.

Before:

js: 12033
wasm: 140

After:

js: 11508
wasm: 2834

**Comment by sbc100** [comment]
I wonder if there are folks out there who would notice this regression.   It seems it would take both conditions to be true:

1. They don't have strtol in their codebase
2. The do include socket code in their codebase  (either they have not pruned out or they are using the proxy socket code).

For any codebase that already *does* include strtol I think this change should be an improvement since it will save on JS codesize.

Assuming we decide we don't want to take this hit perhaps a better place to cut to JS would be strtol itself.  If we did at that level the 2k saving would apply to a lot more folks and we could remove this custom inet_aton without a code size hit for anyone.

**Comment by sbc100** [comment]
Hmm looks like we chose to move from JS to native strtol back in 492b230a84c5cd39585d6c06edeb2bca7709abaa

**Comment by juj** [comment]
Another option is to implement a smaller `strtol` in C/C++: musl adding dependency to the filesystem and FILE handles with `strtox` is stellar. Reducing a specific problem to a general problem and then solving it with an implementation that pulls in heavy guns is a good recipe for code size bloat.

The decision in 2014 was done without analysing code size that much, but to reduce JS-Wasm boundary transitions, which were identified to be a source of performance issues back then.

**Comment by kripken** [comment]
The unfortunate thing seems to be that `strtol` uses a FILE internally. Probably to be able to save code with `__intscan` which is used for filesystem things too. Making an `__intscan` variant that works on a buffer would make this a lot smaller, as @juj suggests.

I think this is rare enough that it's to land as is, with an issue filed for that size optimization eventually.

**Comment by sbc100** [comment]
> The unfortunate thing seems to be that `strtol` uses a FILE internally. Probably to be able to save code with `__intscan` which is used for filesystem things too. Making an `__intscan` variant that works on a buffer would make this a lot smaller, as @juj suggests.
> 
> I think this is rare enough that it's ok to land as is, with an issue filed for that size optimization eventually.

Its ok I think I can wait on this until we have a better `strtol`

**Comment by sbc100** [comment]
I implemented a smaller strtol: #13523.  Once that lands I think we can land this..

**Comment by sbc100** [comment]
With the new small strtol implementation the results are much better:

```
-rw-r----- 1 sbc primarygroup 11508 Feb 18 07:41 new.js
-rwxr-x--- 1 sbc primarygroup  1057 Feb 18 07:41 new.wasm
-rw-r----- 1 sbc primarygroup 12033 Feb 18 07:42 old.js
-rwxr-x--- 1 sbc primarygroup    94 Feb 18 07:42 old.wasm
```

We save 525 of JS but it costs 963 bytes of wasm.    

Overall I think this is a good change now.    If we want to micro-optimize stuff but doing it JS then we should probably start with strtol itself rather than inet_addr.

**Review by kripken — APPROVED** [review]
lgtm after the strol PR lands.

**Pull request merged**
